File: app/src/main/java/com/wavesplatform/wallet/crypto/PrivateKeyAccount.java
package com.wavesplatform.wallet.crypto;

import com.google.common.primitives.Bytes;
import com.google.common.primitives.Ints;

public class PrivateKeyAccount {

    private byte[] publicKey;
    private byte[] privateKey;

    public PrivateKeyAccount(byte[] seed) {
        byte[] hashedSeed = Sha256.hash(Hash.secureHash(Bytes.concat(Ints.toByteArray(0), seed)));

        privateKey = CryptoProvider.get().generatePrivateKey(hashedSeed);
        publicKey = CryptoProvider.get().generatePublicKey(privateKey);
    }

    public byte[] getPublicKey() {
        return publicKey;
    }

    public byte[] getPrivateKey() {
        return privateKey;
    }

    public String getPublicKeyStr() {
        return Base58.encode(getPublicKey());
    }
}

File: wavesplatform/src/main/java/com/wavesplatform/sdk/crypto/PrivateKeyAccount.kt
/*
 * Created by Eduard Zaydel on 1/4/2019
 * Copyright © 2019 Waves Platform. All rights reserved.
 */

package com.wavesplatform.sdk.crypto

import com.google.common.primitives.Bytes
import com.google.common.primitives.Ints

class PrivateKeyAccount(seed: ByteArray) {

    val publicKey: ByteArray
    val privateKey: ByteArray

    val publicKeyStr: String
        get() = Base58.encode(publicKey)

    val privateKeyStr: String
        get() = Base58.encode(privateKey)

    init {
        val hashedSeed = WavesCrypto.sha256(WavesCrypto.keccak(Bytes.concat(Ints.toByteArray(0), seed)))
        privateKey = CryptoProvider.get().generatePrivateKey(hashedSeed)
        publicKey = CryptoProvider.get().generatePublicKey(privateKey)
    }
}
